create_infotexts.py

```diff
@@ -381,9 +374,114 @@ def process_depicted_place(self, place_string, places_mapping):
             self.meta_cats.append("Media_contributed_by_SMVK_without_mapped_place_value")
             place_as_wikitext = place_string
 
-
         self.data["depicted_place"] = place_as_wikitext
 
+    def generate_list_of_stripped_keywords(self, keyword_string):
+        """
+        Transform string of keywords from column <Nyckelord> to list of keywords.
+
+        :param keyword_string: String from column <Nyckelord.
+        :return: list of keywords.
+        """
+        keywords_list = keyword_string.split(", ")
+        if "Svenska Cypernexpeditionen" in keywords_list:
+            keywords_list.remove("Svenska Cypernexpeditionen")
+
+        self.data["Nyckelord"] = keywords_list
+
+    @staticmethod
+    def remove_svenska_cypernexpedition_from_description(description):
+        """
+        Remove the string "Svenska Cypernexpeditionen" from description, since it's in all descriptions.
+
+        :param description: string containg the field value for column <Beskrivning> plus potential enrichments.
+        :return: String with "Svenska Cypernexpeditionen" removed.
+        """
+        new_string = re.sub(" Svenska Cypernexpeditionen\.?", "", description)
+
+        return new_string
+
+    def enrich_description_field(self, item):
+        """
+        Try to add keywords and regional information to description.
+
+        :param item: dictionary containing metadata for one image.
+        :return: string representing altered or unaltered description.
+        """
+        description = self.remove_svenska_cypernexpedition_from_description(item["Beskrivning"])
+
+        if not description.endswith("."):
+            description += "."
+
+        # step 1 in enrichment process
+        description_with_region = self.process_region_addition_to_description(
+            description,
+            item["Region, foto"],
+            item["Land, foto"]
+        )
+
+        # step 2 in enrichment process
+        description_with_keywords = self.process_keyword_addition_to_description(
+            description_with_region
+        )
+
+        # Step 3 in enrichment process
+
+        batch_description = ". Svenska Cypernexpeditionen 1927-1931."
+        full_description = description_with_keywords.rstrip(".") + batch_description
+
+        self.data["enriched_description"] = full_description
+
+    def process_keyword_addition_to_description(self, description_str):
```

Comment:
Hm. Yes, that is definately a more elegant solution. 🏛  I'll do it that way instead (without the <br> though, since I want to place it all inline for esthetical reasons. Done.

I'll create separate logic to enrich filenames.